src/Api/Orbit.cpp

```diff
@@ -101,20 +100,25 @@ static void orbit_api_initialize_v0(uint64_t address) {
   api->track_uint64 = &orbit_api_track_uint64;
   api->track_float = &orbit_api_track_float;
   api->track_double = &orbit_api_track_double;
-  api->initialized = 1;
   std::atomic_thread_fence(std::memory_order_release);
-  api->active = 1;
+  api->initialized = 1;
 }
 
-void orbit_api_initialize(uint64_t address, uint64_t api_version) {
-  LOG("Initializing Orbit API at address %#x, version %u", address, api_version);
+// The "orbit_api_enable" function is called remotely by OrbitService on every capture start for all
+// api function tables. It is also called on every capture stop to disable the api so that the api
+// calls early out at the call site.
+void orbit_api_enable(uint64_t address, uint64_t api_version, bool enabled) {
+  LOG("%s Orbit API at address %#x, version %u", enabled ? "Enabling" : "Disabling", address,
+      api_version);
   if (api_version > kOrbitApiVersion) {
     ERROR(
         "Orbit API version in tracee (%u) is newer than the max supported version (%u). "
         "Some features will be unavailable.",
         api_version, kOrbitApiVersion);
   }
 
-  if (api_version >= 0) orbit_api_initialize_v0(address);
+  orbit_api_v0* api_v0 = absl::bit_cast<orbit_api_v0*>(address);
+  orbit_api_initialize_v0(api_v0);
+  api_v0->enabled = enabled;
```

Comment:
As in the other comment, now you have this assignment following `api->initialized = 1` in `orbit_api_initialize_v0`. Are there new implications regarding atomicity here?

Reply:
We do have atomicity but there could have been an ordering issue, which I don't think we have. I see "enabled" as a global toggle whose value is only valid if "initialized' is 1, in which case we know that all function pointers have been written.

Reply:
Yeah, with "atomicity" I meant the whole package, including ordering.